core/src/test/java/org/apache/iceberg/rest/TestRESTCatalog.java

```diff
@@ -2288,6 +2289,34 @@ public void multipleDiffsAgainstMultipleTablesLastFails() {
     assertThat(schema2.columns()).hasSize(1);
   }
 
+  @Test
+  public void testPaginationForListApis() {
```

Comment:
I think this should be testing pagination for namespaces and tables separately. Additionally, we need to test pagination of views in `TestRESTViewCatalog`

Reply:
will fix and add test in `TestRESTViewCatalog`